src/components/activitylog.js

```diff
@@ -84,17 +77,8 @@ function reloadData(instance, elem, apiClient, startIndex, limit) {
     })).then(function (result) {
         elem.setAttribute('data-activitystartindex', startIndex);
         elem.setAttribute('data-activitylimit', limit);
-        if (!startIndex) {
-            const activityContainer = dom.parentWithClass(elem, 'activityContainer');
-
-            if (activityContainer) {
-                if (result.Items.length) {
-                    activityContainer.classList.remove('hide');
-                } else {
-                    activityContainer.classList.add('hide');
-                }
-            }
-        }
+        const activityContainer = dom.parentWithClass(elem, 'activityContainer');
+        activityContainer?.classList.toggle('hide', result.Items.length === 0 && startIndex === 0);
```

Comment:
Sorry if it wasn't clear, this is what I meant:

```suggestion
        if (startIndex === 0) {
            dom.parentWithClass(elem, 'activityContainer')?.classList
                .toggle('hide', result.Items.length === 0 && startIndex === 0);
        }
```

Reply:
Is the if statement around the class toggle needed? Since it's checking for startIndex === 0 in the toggle statement

Reply:
I'd say yes to avoid calling `parentWithClass`.

Reply:
This is covered by may suggested change I think... it's basically updating the visibility of the container when the index is `0`.

Reply:
@dmitrylyzo do you agree that my recommended change should work here?

Reply:
> @dmitrylyzo do you agree that my recommended change should work here?

In terms of refactor, it is good because the original logic is preserved.
And probably the only correct one. See below.

> What if we get `result.Items.length: 0` with `startIndex: 0` on update (server notification)? We should probably hide the container anyway. 🤔 In `finally`?

I just realized I said the opposite of what I meant to say. :sweat_smile: 
What if we get `result.Items.length === 0` with **`startIndex !== 0`** on update (server notification)?

I tried this (with the original logic):
1. Remove `hide` from the "Alerts" container (it is usually empty).
2. Change `data-activitystartindex` of its child list to `1`. _Think of this as using pagination._
3. Run music in another tab -> logs are going to be updated.
4. "Alerts" container remains visible.

The original logic (which will be preserved by Bill's suggestion) changes visibility only at the start (`startIndex === 0`) and keeps the container visible if there are no more items.

`result.Items.length === 0 && startIndex === 0` will **show** the block just because `startIndex !== 0` - even worse.

Anyway, these blocks don't have pagination and I think we just need to not break what we have. So I'm in favor of leaving `if (startIndex === 0)`.